### PR TITLE
Fix for issue#180. Following are done as part of fix:

### PR DESCRIPTION
**Pull request by SharifuddinAmin**
- Modified DetermineDelayTime() formula to compute delay. The computed delay would be between 0 to default 90% maximum_delay_time sent by requester
    - Modified HandleReceivedListIdentityCommandUdp() & ManageEncapsulationMessages to store and send delayed ListIdentity response message respectively

### CODE REVIEW

**Comment by CapXilinx** [comment]
Hi,

thank you for the fix. Most likely I will not be able to review it until the weekend or Monday.

Best regards,
Martin

**Comment by CapXilinx** [comment]
Hi,

I started checking your fix with the CT tool, and it looks good. Unfortunately my test setup itself seems to have some issues, so I will need some additional days to fix it and then recheck the code.

Best regards,
Martin

**Comment by SharifuddinAmin** [comment]
Hi Martin,

Thanks!
Please take your time, let me know if it works or if something needs to be
changed.

Best Regards,
Sharifuddin Amin

**Comment by SharifuddinAmin** [comment]
> Hi,
> 
> I started checking your fix with the CT tool, and it looks good. Unfortunately my test setup itself seems to have some issues, so I will need some additional days to fix it and then recheck the code.
> 
> Best regards,
> Martin

Hi Martin,

Wondering if you get time to get test setup working and check the code changes done by me.

Thanks & Regards,
Sharifuddin Amin

**Comment by CapXilinx** [comment]
Hi,

sorry for the delay. I am currently reviewing it. I currently investigate an CT issue. It reports that the TCP connection should be closed due to CIP connection timeout, but it isn't. Not sure where it comes from, but I wanted to check before merging.

File: source/src/enet_encap/encap.c [hunk 1]
```diff
@@ -523,7 +532,11 @@ void DetermineDelayTime(const EipByte *const buffer_start,
   } else if (kListIdentityMinimumDelayTime > maximum_delay_time) {       /* if maximum_delay_time is between 1 and 500ms set it to 500ms */
     maximum_delay_time = kListIdentityMinimumDelayTime;
   }
-  delayed_message_buffer->time_out = (maximum_delay_time * rand() ) / RAND_MAX;      /* Sets delay time between 0 and maximum_delay_time */
+  /* Sets delay time between 0 and 90% of maximum_delay_time */
+  /* 90% is taken to have some room for response to reach the requester 
+   * within maximum_delay_time sent by requester
+   */
+  delayed_message_buffer->time_out = rand() % ((maximum_delay_time * kListIdentityDelayXmitBuffer)/100);
```
Comment (CapXilinx) [inline source/src/enet_encap/encap.c]: I am not sure about this factor. I think the scanner should cope with the possible additional delay from the response, and not the adapter. Also, although I am not completely sure if this is the problem, I sometimes get an additional timeout error for class 3 connections in the CT tool, which I never had before.

**Pull request merged**